### ISSUE TITLE
Decomposition Network

### ISSUE THREAD

**Issue by jgy0** [jinyeying/night-enhancement#37]
The article talks about decomposing G, L, and R through three networks.  is it  decomposition2 ?   what is the difference on the three decomposition?  decomposition1,decomposition2,decomposition3

**Comment by jgy0** [comment]
Using demo_all is faster, but the results are not good. On the other hand, using demo_separation is too slow, taking a long time for even a single image. What exactly did the author describe in the paper? Now, trying to use this to train my own data is very difficult. Could it be that the code is incomplete?

**Comment by jinyeying** [comment]
> The article talks about decomposing G, L, and R through three networks. is it decomposition2 ? what is the difference on the three decomposition? decomposition1,decomposition2,decomposition3

Hi, sorry for the late reply.  

The article talks about decomposing G, L, and R through three networks. Is it decomposition2? 
Answer: No, it is [Decomposition3](https://github.com/jinyeying/night-enhancement/blob/main/decomposition_code/demo_decomposition.m)

Difference on the three decompositions?  Answer: You can see the explanation images: 
1. demo1 works for red light-effects since the Color Constancy Loss Eq4 in our paper (https://github.com/jinyeying/night-enhancement/blob/main/teaser/demo1.png), 
2. demo2  is DoubleDIP-based, works for white flow (https://github.com/jinyeying/night-enhancement/blob/main/teaser/demo2.png)
3. demo3 is decomposition/ prior-based (https://github.com/jinyeying/night-enhancement/blob/main/teaser/decomposition.png), 

In the paper, it is
For the light-effects layer, we use a light-effects map Gi, computed using the relative smoothness technique [22]. This is extracted using the second-order Laplacian filter from the input image, since light effects are smooth variations. We define the loss function for the initialization step as:
Linit = |G − Gi|1 + |L − Li|1

Demo3 are the results for Gi and  Li, which are the ground truth for the decomposition network.

**Comment by jinyeying** [comment]
> Using demo_all is faster, but the results are not good. On the other hand, using demo_separation is too slow, taking a long time for even a single image. What exactly did the author describe in the paper? Now, trying to use this to train my own data is very difficult. Could it be that the code is incomplete?

Hi, sorry for the late reply. 

The idea is very simple, demo_separation is too slow, but has good results, use these results as (sudo-)ground truth to train your network; it is the decomposition guidance.

In the paper, it is
For the light-effects layer, we use a light-effects map Gi, computed using the relative smoothness technique [22]. This is extracted using the second-order Laplacian filter from the input image, since light effects are smooth variations. We define the loss function for the initialization step as:
Linit = |G − Gi|1 + |L − Li|1
